internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -192,6 +190,9 @@ type Coordinator struct {
 	state coordinatorState
 }
 
+// ErrFatalCoordinator is returned when a coordinator sub-component returns an error, as opposed to a simple context-cancelled.
+var ErrFatalCoordinator = fmt.Errorf("fatal error in coordinator")
```

Comment:
Nitpicking: if you don't need to format or wrap errors, the common way to create a new error is to use `errors.New`

```suggestion
var ErrFatalCoordinator = errors.New("fatal error in coordinator")
```